xtask/src/codegen/gen_unstable_future_descriptor.rs

```diff
@@ -0,0 +1,62 @@
+//! Generates descriptors structure for unstable feature from Unstable Book
+
+use crate::codegen::update;
+use crate::codegen::{self, project_root, Mode, Result};
+use quote::quote;
+use std::fs;
+use std::process::Command;
+use walkdir::WalkDir;
+
+pub fn generate_unstable_future_descriptor(mode: Mode) -> Result<()> {
+    let path = project_root().join(codegen::STORAGE);
+    fs::create_dir_all(path.clone())?;
+
+    Command::new("git").current_dir(path.clone()).arg("init").output()?;
+    Command::new("git")
+        .current_dir(path.clone())
+        .args(&["remote", "add", "-f", "origin", codegen::REPOSITORY_URL])
+        .output()?;
+    Command::new("git")
+        .current_dir(path.clone())
+        .args(&["sparse-checkout", "set", "/src/doc/unstable-book/src/"])
+        .output()?;
```

Comment:
https://www.git-scm.com/docs/git-sparse-checkout

> THIS COMMAND IS EXPERIMENTAL. ITS BEHAVIOR, AND THE BEHAVIOR OF OTHER COMMANDS IN THE PRESENCE OF SPARSE-CHECKOUTS, WILL LIKELY CHANGE IN THE FUTURE.

Reply:
Well, yes, but how else can I get a separate directory and check it? I am really not good enough at git, so any ideas are welcome

Reply:
A bit more complex, but you could use the github api. This also completely avoids the clone.

Reply:
Ok, it's seems reasonable, I try it.

Reply:
> A bit more complex, but you could use the github api. This also completely avoids the clone.
We need a token, what about it?

Reply:
As far as I know you can access most apis without token, just with much lower rate limits.

https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api#authentication

> Unauthenticated clients can make 60 requests per hour. To get more requests per hour, we'll need to authenticate. In fact, doing anything interesting with the GitHub API requires authentication.

Reply:
This is the problem, you can do a check, but  you cannot integrate this without a token in CI

Reply:
CI can use `${{ secrets.GITHUB_TOKEN }}`: https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token

Reply:
> CI can use `${{ secrets.GITHUB_TOKEN }}`: https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token

thank you for help

Reply:
mkladov said that this implementation is contrary to the goals